parse_snyk_output.cjs

```diff
@@ -0,0 +1,178 @@
+const fs = require('fs');
+const JiraClient = require('jira-client');
+const rp = require('request-promise');
+const fetch = require('node-fetch');
+const base64 = Buffer.from(`${process.env.JIRA_USERNAME}:${process.env.JIRA_API_TOKEN}`).toString('base64');
+
+
+console.log("JIRA_BASE_URL:", process.env.JIRA_BASE_URL);
+
+const jira = new JiraClient({
+  protocol: 'https',
+  host: process.env.JIRA_BASE_URL,
+  username: process.env.JIRA_USER_EMAIL,
+  password: process.env.JIRA_API_TOKEN,
+  apiVersion: '2',
+  strictSSL: true,
+});
+
+
+function parseSnykOutput(inputData) {
+  let vulnerabilities = [];
+ 
+  console.log(inputData);
+
+  if (inputData) {
+    try {
+      const data = JSON.parse(inputData);
+      for (const project of data) {
+        vulnerabilities = vulnerabilities.concat(project.vulnerabilities);
+      }
+
+      console.log(vulnerabilities);
+
+    } catch (error) {
+      console.error('Error parsing Snyk output:', error);
+      vulnerabilities = parseNonJsonData(inputData);
+    }
+  }
+
+  return vulnerabilities;
+}
+
+
+function parseNonJsonData(inputData) {
+  let vulnerabilities = [];
+
+  // Custom logic to parse non-JSON inputData
+  const defaultTitle = 'Vulnerability Detected';
+
+  vulnerabilities.push({
+    title: defaultTitle,
+    description: `Non-JSON output from Snyk:\n\n${inputData}`
+  });
+
+  return vulnerabilities;
+}
+
+
+async function createJiraTicket(vulnerability) {
+   // DEFAULT DAYS  set to 60 adjust as needed
+   const sixtyDaysAgo = new Date(Date.now() - 60 * 24 * 60 * 60 * 1000).toISOString().split('T')[0];
+  //const today = new Date().toISOString().split('T')[0];
+
+  // JQL query to check ticket given title from the past given days
+  let jqlQuery = `project = "${process.env.JIRA_PROJECT_KEY}" AND summary ~ "MCR - ${vulnerability.title}" AND created >= ${sixtyDaysAgo}`;
+  let searchResult = await jira.searchJira(jqlQuery);
+
+
+  if (searchResult.issues && searchResult.issues.length > 0) {
+    for (const issue of searchResult.issues) {
+      if (issue.fields.status.name !== "Closed" && issue.fields.status.name !== "Cancelled") {
```

Comment:
Idea: JQL supports status conditions. Instead of checking issue status here we could include the search clause `AND status NOT IN ("Closed", "Cancelled")`.